corehq/form_processor/submission_post.py

```diff
@@ -571,3 +514,104 @@ def notify_submission_error(instance, message, exec_info=None):
         notify_exception(request, message, details=details, exec_info=exec_info)
     else:
         logging.error(message, exc_info=sys.exc_info(), extra={'details': details})
+
+
+class SubmissionPostFormProcessor(object):
+    def __init__(self, *, domain, instance, attachments, auth_context, submit_ip, path, openrosa_headers,
+                 last_sync_token, received_on, date_header, app_id, build_id, partial_submission):
+        self.domain = domain
+        self._instance = instance
+        self.attachments = attachments or {}
+        self.auth_context = auth_context or DefaultAuthContext()
+        self.submit_ip = submit_ip
+        self.path = path
+        self.openrosa_headers = openrosa_headers
+        self.last_sync_token = last_sync_token
+        self.received_on = received_on
+        self.date_header = date_header
+        self.app_id = app_id
+        self.build_id = build_id
+        self.partial_submission = partial_submission
+
+    def process_form(self):
+        self.pre_process_form()
+        result = process_xform_xml(self.domain, self.get_instance(), self.attachments, self.auth_context.to_json())
+        self.post_process_form(result.submitted_form)
```

Comment:
how about this? I think it make the flow a bit clearer. You can also then remove `get_instance`
```suggestion
        instance = self.pre_process_form()
        result = process_xform_xml(self.domain, instance, self.attachments, self.auth_context.to_json())
        self.post_process_form(result.submitted_form)
```

Reply:
as commented above, we can't remove `get_instance`.

Additionally, just curious, and to understand better,
If there is an accessor added specifically for an attribute, would it make sense to return it from one of the methods and then pass that around?

also, that makes it challenging, when the method that overrites it, https://github.com/dimagi/commcare-icds/pull/54/files#diff-3fbda5effcae7a41cec587acf46e8011R12 should always return super() call. As a safety, i made that overrite return whatever received from super() call even though the method here does not return anything. That forces that calling super() was the last step. is there a general guideline for methods that are allowed to be orverritten, to return/not return anything.
(i hope i explained that well, happy to rephrase that if needed)

Reply:
I don't see why this is so complex:
```
class DefaultFormProcessor:

    def pre_process_form(self):
        return self._instance

class ICDSFormProcessor:
    def pre_process_form(self):
        instance = super().pre_process_form()
        instance = self._extract_vault_values(instance)
        return instance
```

Reply:
https://github.com/dimagi/commcare-hq/pull/28426/commits/4d81fcac48dd5c536d32c42fa91a4acd664cd4ef